### ISSUE TITLE
Is Grok fully supported?

### ISSUE THREAD

**Issue by m-kn** [logzio/logzio-lambda-extensions#19]
Hi, it's me _again_

This [Grok Tutorial](https://logz.io/blog/logstash-grok/), which is linked in the [documentation](https://github.com/logzio/logzio-lambda-extensions/tree/main/logzio-lambda-extensions-logs#parsing-logs), lists some examples and among them some MongoDB example, which in turn includes `(?<duration>[0-9]+)ms`. Because of that I thought Oniguruma named groups i.e. `(?<name>subexp)` would end up as fields in Logz.io.

Here's an example line of my logs
`2022-03-16 10:33:55 INFO  com.foocorp.image.processing.lambda.ImageProcessingLambda <26d362c3-2d5e-4b88-9d20-ac5f4eb71c91> - handle 1 records`

Using `(?<custom_message>.*)` as my Grok, [grokdebug](http://grokdebug.herokuapp.com/) says I should be getting
```
{
  "custom_message": [
    [
      "2022-03-16 10:33:55 INFO  com.foocorp.image.processing.lambda.ImageProcessingLambda <26d362c3-2d5e-4b88-9d20-ac5f4eb71c91> - handle 1 records"
    ]
  ]
}
```
To test if this works at all I set the following environment variables for my lambda/the logz.io extension
`GROK_PATTERNS` to `{}`
and
`LOGS_FORMAT` to `(?<custom_message>.*)`
and got
```
"message": "2022-03-16 10:33:55 INFO  com.foocorp.image.processing.lambda.ImageProcessingLambda <26d362c3-2d5e-4b88-9d20-ac5f4eb71c91> - handle 1 records\n",
```
in logz.io, which is the fallback, i.e. parse the whole line as a string into `message`. 

To make sure the problem not the `message` field being empty I set
`LOGS_FORMAT` to `%{GREEDYDATA:custom_message}` 
instead and get
```
"custom_message": "2022-03-16 10:43:05 INFO  com.foocorp.image.processing.lambda.ImageProcessingLambda <f071c761-fe5d-44eb-bbb7-b35531e268c6> - handle 1 records"`
```
so that's not the problem.

Back to the previous attempt, I took a look at the extension logs and saw the following error
```
time="2022-03-16T11:00:29Z" level=error msg="error occurred while trying to parse fields. sedning log as a string: error parsing regexp: invalid or unsupported Perl syntax: `(?<`" agent=logzioLogsExtension
```
(typo in "sedning", heh)
which to me doesn't look this is supposed to work at all.

Anyway, my example here would've done the same as the fallback behaviour except put the result into `custom_message`, but this just a simplified example for demonstration purposes. My actual requirement was have some UUID be optional so just `%{UUID:foo}?` as part of my `LOGS_FORMAT` does the trick. This also means some subset of RegEx (Perl?/Not-Oniguruma?) seems to be allowed in `LOGS_FORMAT`.

The TL;DR of all of this above is this: 
**Is Grok really fully supported?** 
- If yes, the named groups not working in `LOGS_FORMAT` looks like a bug to me
- If not, I would like some sort of hint in the documentation, so I'm not trying to use some Grok features and then left wondering and debugging why they're not working

**Comment by mirii1994** [comment]
Hi @m-kn ,
Sorry you encountered difficulties. Thank you for letting us know that there are some issues with the documentation.
This project uses [this go module](https://github.com/vjeantet/grok) for parsing grok patterns, and seems like the Oniguruma syntax is not supported. We'll change our documentation to avoid further confusion.
If I understand correctly, in the example you've given, you want the entire log to appear under the field `custom_message`, right?
If so, then the following configuration will work:
`GROK_PATTERNS ` = `{"custom_message":".*"}`
`LOGS_FORMAT ` = `%{custom_message:custom_message}`

**Comment by mirii1994** [comment]
@m-kn We've updated our documentation, so i'm closing this issue.
If you have any more questions, feel free to re-open this one or open a new issue :)
Thanks again for helping us improve this project!
